devicetypes/smartthings/arrival-sensor-ha.src/arrival-sensor-ha.groovy

```diff
@@ -79,21 +100,37 @@ def beep() {
 }
 
 def parse(String description) {
-    state.lastCheckin = now()
-    handlePresenceEvent(true)
-
-    if (description?.startsWith('read attr -')) {
-        handleReportAttributeMessage(description)
+	log.debug "description: $description"
+	state.lastCheckin = now()
+    if (device.getDataValue("manufacturer") == "SmartThings"){
+    	handlePresenceEvent(true)
     }
-
-    return []
-}
-
-private handleReportAttributeMessage(String description) {
```

Comment:
Can You explain why did You included code from this method into parse() ? It wasn't necessary.

Reply:
Sorry! I will revert this change.